pkg/manager/internal.go

```diff
@@ -349,28 +349,23 @@ func (cm *controllerManager) getLogicalCluster(ctx context.Context, name logical
 	}
 
 	// Create a new cluster.
-	var cfg *rest.Config
+	var cl cluster.Cluster
 	{
 		// TODO(vincepri): Make this timeout configurable.
 		ctx, cancel := context.WithTimeout(ctx, 10*time.Second)
 		defer cancel()
 		var watchErr error
 		if err := wait.PollImmediateUntilWithContext(ctx, 1*time.Second, func(ctx context.Context) (done bool, err error) {
-			cfg, watchErr = cm.logicalAdapter.RESTConfig(name)
+			cl, watchErr = cm.logicalProvider.Get(ctx, name, cm.defaultClusterOptions, cluster.WithName(name))
```

Comment:
Why both name and cluster.WithName(name)?

Reply:
The last one just sets the option for `cluster.New`

Reply:
Happy to rework it, I guess it's a bit weird 😄